api/envoy/config/trace/v2/trace.proto

```diff
@@ -184,6 +184,13 @@ message OpenCensusConfig {
   // https://github.com/grpc/grpc/blob/master/doc/naming.md
   string stackdriver_address = 10;
 
+  // (optional) The call credentials used in gRPC call to production Stackdriver.
+  // Currently only :ref:`sts_service <envoy_v3_api_field_config.core.v3.GrpcService.GoogleGrpc.CallCredentials.sts_service>`
+  // is supported by opencensus tracer. If not provided, or the provided option is not
+  // :ref:`sts_service <envoy_v3_api_field_config.core.v3.GrpcService.GoogleGrpc.CallCredentials.sts_service>`,
+  // the default Google credential will be used as call credential.
+  api.v2.core.GrpcService.GoogleGrpc.CallCredentials stackdriver_call_credentials = 13;
```

Comment:
Shouldn't we just have an `api.v2.core.GrpcService` service to cover where OpenCensus connect to and the options?

Reply:
Opencensus library uses its own network stack, not sure how can I use envoy google gRPC stack to replace that. It provides an mechanism to specify a gRPC stub, but it expects stub of a specific proto (https://github.com/census-instrumentation/opencensus-cpp/blob/c900c4d723ef596f357b0e695e68e2fa725eec90/opencensus/exporters/trace/stackdriver/stackdriver_exporter.h#L40). Will GoogleStub be compatible with that?

On the other hand, Opencensus library is unlikely to utilize all the options provided by the GrpcService. I can only foresee that channel credential might be needed in the future. So adding the whole gRPC service here seems an overkill for me.

Reply:
That is a generated Google gRPC stub from the code? So, you should be able to reuse (might need some refactor) from [here](https://github.com/envoyproxy/envoy/blob/5248a4fb7d4c2a3d1fa151f944d3a63f6b7a06cf/source/common/grpc/google_async_client_impl.cc#L83-L85), i.e. calling `google::devtools::cloudtrace::v2::TraceService::NewStub` on the channel created by these code. Then you don't have to worry about other features added in gRPC not being picked up by OpenCensus.

You can add "only Google gRPC in GrpcService is supported" doc (and validate in code) so `EnvoyGrpc` won't be configured.

Reply:
Done.